app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java

```diff
@@ -157,120 +300,74 @@ public void onTabReselected(TabLayout.Tab tab) {
 
         @Override
         public Fragment getItem(int position) {
-            switch (position) {
-                case 0:
-                    return isSubscriptionsPageOnlySelected() ? new SubscriptionFragment() : getMainPageFragment();
-                case 1:
-                    if(PreferenceManager.getDefaultSharedPreferences(getActivity())
-                            .getString(getString(R.string.main_page_content_key), getString(R.string.blank_page_key))
-                            .equals(getString(R.string.subscription_page_key))) {
-                        return new BookmarkFragment();
-                    } else {
-                        return new SubscriptionFragment();
+            String tabNumber = tabs.get(position);
+
+            if(tabNumber.startsWith(TAB_NUMBER_KIOSK + "\t")) {
```

Comment:
What is tabNumber, and why does it need to be split later?
What would be a typical value it holds?